clang/test/CIR/CodeGen/complex-builtins.cpp

```diff
@@ -0,0 +1,21 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -Wno-unused-value -fclangir -emit-cir %s -o %t.cir
+// RUN: FileCheck --input-file=%t.cir %s -check-prefix=CIR
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -Wno-unused-value -fclangir -emit-llvm %s -o %t-cir.ll
+// RUN: FileCheck --input-file=%t-cir.ll %s -check-prefix=LLVM
+
+void complex_choose_expr() {
+ int _Complex a;
+ int _Complex b;
```

Comment:
maybe have these as input arguments to the function so you don't need to check the allocas?

Reply:
As far as I understand, I think using arguments will end up with allocas and store (argN to alloca), then the same code to load and store to the result alloca, and I will need to check which alloca is used 🤔

Reply:
that's fair, the source code just looks a bit nicer though. I'm fine with this however :)